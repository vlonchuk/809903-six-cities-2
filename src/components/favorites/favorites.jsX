import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import Operation from './../../reducer/operation/operation.js';
import PageHeader from './../page-header/page-header.jsx';
import PlacesList from './../places-list/places-list.jsx';
import {Link} from "react-router-dom";

class Favorites extends PureComponent {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    this.props.loadFavorites();
  }

  render() {
    return <div className="page">
      <PageHeader user={this.props.user} />
      <div className="page__favorites-container container">
        {
          this.props.properties.length > 0 ?
            this.renderFull() :
            this.renderEmpty()
        }
      </div>
      {this.renderFooter()}
    </div>;
  }

  renderFull() {
    return <PlacesList forCity={true} key="PlacesList" properties={this.props.properties}/>;
  }

  renderEmpty() {
    return <section className="favorites favorites--empty">
      <h1 className="visually-hidden">Favorites (empty)</h1>
      <div className="favorites__status-wrapper">
        <b className="favorites__status">Nothing yet saved.</b>
        <p className="favorites__status-description">Save properties to narrow down search or plan yor future trips.</p>
      </div>
    </section>;
  }

  renderFooter() {
    return <footer className="footer">
      <Link to="/" className="footer__logo-link">
        <img className="footer__logo" src="/img/logo.svg" alt="6 cities logo" width="64" height="33"></img>
      </Link>
    </footer>;
  }
}

Favorites.propTypes = {
  user: PropTypes.object,
  properties: PropTypes.array.isRequired,
};

const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  user: state.user,
  properties: state.favorites,
});

const mapDispatchToProps = (dispatch) => ({
  loadFavorites: () => {
    dispatch(Operation.loadFavorites());
  },
});

const FavoritesWrapped = connect(mapStateToProps, mapDispatchToProps)(Favorites);
export default FavoritesWrapped;
