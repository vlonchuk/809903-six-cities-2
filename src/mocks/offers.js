const offers = [
  {
    id: `prop-1`,
    city: `Amsterdam`,
    caption: `Beautiful & luxurious apartment at great location`,
    imgSrc: `img/apartment-01.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 120,
    priceText: `night`,
    rating: 50,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.85309666406198
    },
  },
  {
    id: `prop-2`,
    city: `Amsterdam`,
    caption: `Wood and stone place`,
    imgSrc: `img/room.jpg`,
    type: `Private room`,
    priceCurrency: `€`,
    priceValue: 80,
    priceText: `night`,
    rating: 80,
    coor: {
      latitude: 52.369553943508,
      longitude: 4.85309666406198
    },
  },
  {
    id: `prop-3`,
    city: `Amsterdam`,
    caption: `Canal View Prinsengracht`,
    imgSrc: `img/apartment-02.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 132,
    priceText: `night`,
    rating: 100,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
  },
  {
    id: `prop-4`,
    city: `Amsterdam`,
    caption: `Nice, cozy, warm big bed apartment`,
    imgSrc: `img/apartment-03.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 180,
    priceText: `night`,
    rating: 65,
    coor: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
  },
  {
    id: `prop-5`,
    city: `Paris`,
    caption: `Canal View Prinsengracht`,
    imgSrc: `img/apartment-02.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 132,
    priceText: `night`,
    rating: 90,
    coor: {
      latitude: 52.3909553943508,
      longitude: 4.929309666406198
    },
  },
  {
    id: `prop-6`,
    city: `Paris`,
    caption: `Nice, cozy, warm big bed apartment`,
    imgSrc: `img/apartment-03.jpg`,
    type: `Aparment`,
    priceCurrency: `€`,
    priceValue: 180,
    priceText: `night`,
    rating: 40,
    coor: {
      latitude: 52.3809553943508,
      longitude: 4.939309666406198
    },
  },
];

export const getPropertiesByCity = (city) => {
  return offers.filter((el) => el.city === city)
               .slice();
};

export default offers;
