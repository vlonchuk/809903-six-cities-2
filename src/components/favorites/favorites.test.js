import React from 'react';
import renderer from 'react-test-renderer';
import {Favorites} from './favorites';
import offers from './../../mocks/offers.js';
import {BrowserRouter as Router} from 'react-router-dom';

it(`Favorites correctly renders after relaunch`, () => {
  const tree = renderer
    .create(
        <Router>
          <Favorites
            properties={offers}
            loadFavorites={jest.fn()}
          />
        </Router>
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
