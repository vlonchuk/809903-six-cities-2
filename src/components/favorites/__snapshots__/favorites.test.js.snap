// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites correctly renders after relaunch 1`] = `null`;
